libraries/chain/controller.cpp

```diff
@@ -1440,7 +1440,7 @@ struct controller_impl {
          initialize_blockchain_state(genesis); // sets head to genesis state
 
          if( !forkdb.head() ) {
-            forkdb.reset( *forkdb.chain_head );
+            forkdb.reset_root( *forkdb.chain_head );
```

Comment:
Why not call `fork_db_reset_root_to_head()` here?

We have `forkdb.reset_root( *head )`, and `forkdb.reset_root( *forkdb.chain_head );` which is equivalent to `fork_db_reset_root_to_head()`. Should `fork_db_reset_root_to_head()` be renamed to a more clearer name?

Reply:
Renamed to `fork_db_reset_root_to_chain_head`. This location can't use that because it is inside a lambda that is using the specific `forkdb` type. This is being refactored in the PR for #2141 as `chain_head` is moving from `fork_database` back to `controller_impl`.